apps/playground/rom_ui.js

```diff
@@ -0,0 +1,104 @@
+// Visualize the compiled ROM image of a uFork program.
+
+/*jslint browser */
+
+import hexdump from "https://ufork.org/lib/hexdump.js";
+import make_ui from "./ui.js";
+import dom from "./dom.js";
+import theme from "./theme.js";
+
+const rom_ui = make_ui("rom-ui", function (element, {
+    buffer,
+    format = "memfile",
+    bitwidth = 16
+}) {
+    const shadow = element.attachShadow({mode: "closed"});
+    const style = dom("style", `
+        :host {
+            display: flex;
+            flex-direction: column;
+            color: white;
+        }
+        ${theme.monospace_font_css}
+        rom_dump {
+            font-family: ${theme.monospace_font_family}, monospace;
+            font-size: 17px;
+            line-height: 1.3;
+            white-space: pre;
+            min-height: 60px;
+            overflow: auto;
+            scrollbar-color: ${theme.gray} transparent;
+            caret-color: white;
+            color: #3F0;
+            flex: 1 1;
+        }
+        rom_controls {
+            display: flex;
+            justify-content: flex-end;
+            margin-top: 6px;
+        }
+        rom_controls > * {
+            margin-left: 4px;
+        }
+    `);
+    const dump_element = dom("rom_dump", {
+        contentEditable: "true",
```

Comment:
Seems like this should be "false". Isn't this a read-only view?

Reply:
The setting the `contentEditable` property to `true` imbues the element with `<textarea>`-like qualities - specifically, the ability to hold a cursor and respond correctly to the "select all" command - without styling it like a `<textarea>`. You are right, ideally it would be readonly, but that's extra work in this case.

Reply:
I've set `contentEditable: false` and it seems to be behaving correctly. Both range-select and select-all do what I expect, and the content is read-only.

Reply:
Nice, I didn't know it worked like that.